finality-aleph/src/aggregator/mod.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{metrics::Checkpoint, network::DataNetwork, Metrics};
+use aleph_bft::{MultiKeychain, Recipient, Signable};
+use codec::Codec;
+use futures::{channel::mpsc, StreamExt};
+use log::{debug, trace, warn};
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    fmt::Debug,
+    marker::PhantomData,
+};
+
+mod multicast;
+
+pub use multicast::{Hash, Multicast, Multisigned, NetworkData as RmcNetworkData, SignableHash};
+
+/// A type encapsulating three different results of the `process_network_messages` method
+pub(crate) enum AggregatorError {
+    NetworkChannelClosed,
+}
+
+/// A wrapper around an `rmc::Multicast` returning the signed hashes in the order of the [`Multicast::start_multicast`] calls.
+pub(crate) struct BlockSignatureAggregator<
```

Comment:
personally, I do prefer light `mod.rs` files, thus, I would suggest placing this structure in a dedicated file and reexport it from here; for me, rather the content of `multicast.rs` should go here, but actually, it can stay where it is now as well

Reply:
I thought about that but it comes down to the hardest problem of them all: naming the new module :joy:
Any suggestions?

Reply:
I see that this is resolved in a good way :)